hack/make-rules/test.sh

```diff
@@ -32,28 +32,21 @@ export KUBE_CACHE_MUTATION_DETECTOR
 KUBE_PANIC_WATCH_DECODE_ERROR="${KUBE_PANIC_WATCH_DECODE_ERROR:-true}"
 export KUBE_PANIC_WATCH_DECODE_ERROR
 
-kube::test::find_dirs() {
+kube::test::find_packages() {
   (
     cd "${KUBE_ROOT}"
-    find -L . -not \( \
-        \( \
-          -path './_artifacts/*' \
-          -o -path './_output/*' \
-          -o -path './cmd/kubeadm/test/*' \
-          -o -path './contrib/podex/*' \
-          -o -path './release/*' \
-          -o -path './target/*' \
-          -o -path './test/e2e/e2e_test.go' \
-          -o -path './test/e2e_node/*' \
-          -o -path './test/e2e_kubeadm/*' \
-          -o -path './test/integration/*' \
-          -o -path './third_party/*' \
-          -o -path './staging/*' \
-          -o -path './vendor/*' \
-        \) -prune \
-      \) -name '*_test.go' -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
-
-    find ./staging -name '*_test.go' -not -path '*/test/integration/*' -prune -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
+
+    local -a workspace_module_patterns
+    kube::util::read-array workspace_module_patterns < <(go work edit -json | jq -r '.Use[].DiskPath + "/..."')
+
+    go list -find "${workspace_module_patterns[@]}" \
```

Comment:
```suggestion
    go list -f '{{if or (gt (len .TestGoFiles) 0) (gt (len .XTestGoFiles) 0)}}{{.ImportPath}}{{end}}' -find "${workspace_module_patterns[@]}" \
```

see https://github.com/kubernetes/kubernetes/pull/127480#issuecomment-2366829138